src/FFmpegWriter.cpp

```diff
@@ -1361,23 +1352,51 @@ void FFmpegWriter::open_video(AVFormatContext *oc, AVStream *st) {
 
 #if IS_FFMPEG_3_2
 	if (hw_en_on && hw_en_supported) {
-		video_codec->max_b_frames = 0;        // At least this GPU doesn't support b-frames
-		video_codec->pix_fmt   = hw_en_av_pix_fmt;
-		video_codec->profile = FF_PROFILE_H264_BASELINE | FF_PROFILE_H264_CONSTRAINED;
-		av_opt_set(video_codec->priv_data,"preset","slow",0);
-		av_opt_set(video_codec->priv_data,"tune","zerolatency",0);
-		av_opt_set(video_codec->priv_data, "vprofile", "baseline", AV_OPT_SEARCH_CHILDREN);
+		switch (video_codec->codec_id) {
+			case AV_CODEC_ID_H264:
+				video_codec->max_b_frames = 0;        // At least this GPU doesn't support b-frames
+				video_codec->pix_fmt   = hw_en_av_pix_fmt;
+				video_codec->profile = FF_PROFILE_H264_BASELINE | FF_PROFILE_H264_CONSTRAINED;
+				av_opt_set(video_codec->priv_data, "preset", "slow", 0);
+				av_opt_set(video_codec->priv_data, "tune", "zerolatency", 0);
+				av_opt_set(video_codec->priv_data, "vprofile", "baseline", AV_OPT_SEARCH_CHILDREN);
+				break;
+			case AV_CODEC_ID_HEVC:
+				// must be zero (at least on AMDGPU on Linux 8/2019)
+				video_codec->bit_rate = 0;
+
+				//TODO: this can be made configurable (25 default)
+				//av_opt_set(video_codec->priv_data,"qp","27",0);
+
+				// FYI other hevc_vaapi ffmpeg defaults are this
+				/*
+					video_codec->max_b_frames = 2;
+					video_codec->gop_size = 120;
+					video_codec->i_quant_factor = 1;
+					video_codec->i_quant_offset = 0;
+					video_codec->b_quant_factor = 6.0/5;
+					video_codec->qmin = -1;
+					video_codec->qmax = -1;
+				*/
+				break;
+			default:
+				ZmqLogger::Instance()->AppendDebugMethod("No codec-specific options defined for this codec. HW encoding may fail",
+					"codec_id", video_codec->codec_id);
+				break;
```

Comment:
I'm tempted to say this should  just `throw InvalidOptions("No hardware encoding parameters have been defined for this codec.", path);` or something like that, instead. It doesn't feel like there's any reasonable chance that an unknown, unconfigured codec _would_ successfully be encoded by the hardware, right?

Reply:
actually there is some possibility it may work with some default options (default quality). For example HEVC would work without any config as long as the bitrate were 0. I will do some tests with other codecs on my AMD card out of curiosity.